Config test fails if home dir has .eslintrc file

While i was working on `master` [this config unit test](https://github.com/eslint/eslint/blob/master/tests/lib/config.js#L265) fails if you have a `.eslintrc` file inside your home directory. I think this test was introduced with #6538.

To reproduce just create a `.eslintrc` file inside your HOME directory and then run the tests.

```js
{
	"env" : {
		"es6" : true
	}
}
```

output:

```sh
 11186 passing (48s)
  1 failing

  1) Config getConfig() should throw an error when an invalid path is given:
     AssertionError: expected [Function] to throw an error
      at Function.assert.throws (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\chai\lib\chai\interface\assert.js:1120:52)
      at Context.<anonymous> (C:\Users\gs025879\Documents\webstrom\eslint\tests\lib\config.js:271:20)
      at callFn (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runnable.js:326:21)
      at Test.Runnable.run (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runnable.js:319:7)
      at Runner.runTest (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:422:10)
      at C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:528:12
      at next (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:342:14)
      at C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:352:7
      at next (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:284:14)
      at Immediate._onImmediate (C:\Users\gs025879\Documents\webstrom\eslint\node_modules\mocha\lib\runner.js:320:5)
```

Note: This issue is similar to what was happening here #6206

--- a/tests/lib/config.js
+++ b/tests/lib/config.js
@@ -264,7 +264,11 @@ describe("Config", function() {
 
         it("should throw an error when an invalid path is given", function() {
             var configPath = path.resolve(__dirname, "..", "fixtures", "configurations", "foobaz", ".eslintrc");
-            var configHelper = new Config({cwd: process.cwd()});
+            var homePath = "does-not-exist";
+
+            var StubbedConfig = proxyquire("../../lib/config", { "user-home": homePath });
+
+            var configHelper = new StubbedConfig({cwd: process.cwd()});
 
             sandbox.stub(fs, "readdirSync").throws(new Error());
 

